library/alloc/src/string.rs

```diff
@@ -2298,7 +2298,8 @@ impl ops::IndexMut<ops::RangeToInclusive<usize>> for String {
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
-impl ops::Deref for String {
+#[rustc_const_unstable(feature = "const_deref", issue = "none")]
+impl const ops::Deref for String {
```

Comment:
We should not make this (and `Vec`'s impl) `const`, out of the same reason we didn't make `len` `const fn`: there is no way to construct any interesting (non-empty) `String`s, so all these functions are useless. It was decided that, until we have some sort of const heap, we should not add any new const functions beyond constructors.